### PR TITLE
fix(terraform): Update CKV_AWS_338 message and retention check for 0

### LINKED ISSUE #5009: CKV_AWS_338 does not account for 0 (keep forever) & has insufficient error message
**Describe the issue**
CKV_AWS_338 is a check for the number of retention days for a Cloudwatch log group. However, the actual check source code requires not just being specified, but also to be at least 1 year.

1 - The message "Ensure that CloudWatch Log Group specifies retention days" does not reflect the need to not only be specified, but also be of sufficient length (or provide ref to web page providing more detail on the check)

2 - 0 is a valid value and means that logs are always retained and never expire, which should result in passing the "must retain for at least 1 year" rule.

**Examples**
Please share an example code sample (in the IaC of your choice) + the expected outcomes.

**Version (please complete the following information):**
2.3.212

### PR DESCRIPTION
**Pull request by quixoticmonk**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

This PR addresses the correction of message displayed to users in the issue listed below. 
The current message matches CKV_AWS_66 which doesn't reflect the length of time.

Fixes #5009

### Description
- AWS Security Hub controls update : https://docs.aws.amazon.com/securityhub/latest/userguide/cloudwatch-controls.html#cloudwatch-16
- Related requirements: NIST.800-53.r5 AU-10, NIST.800-53.r5 AU-11, NIST.800-53.r5 AU-6(3), NIST.800-53.r5 AU-6(4), NIST.800-53.r5 CA-7, NIST.800-53.r5 SI-12

### Fix
Updated the message displayed for the rule and included a check for retention of 0 which is allowed per Terraform documentation.
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_group#retention_in_days

## Checklist:

- [X] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/5.Policy Index/all.md
Comment (gruebel) [inline docs/5.Policy Index/all.md:1]: please remove this change, it will be automatically generated

File: docs/5.Policy Index/terraform.md
Comment (gruebel) [inline docs/5.Policy Index/terraform.md:1]: same here

File: checkov/terraform/checks/resource/aws/CloudWatchLogGroupRetentionYear.py [hunk 1]
```diff
@@ -13,7 +14,7 @@ def __init__(self) -> None:
         NIST.800-53.r5 CA-7, NIST.800-53.r5 SI-12
         CloudWatch log groups should be retained for at least 1 year
         """
-        name = "Ensure that CloudWatch Log Group specifies retention days"
+        name = "CloudWatch log groups should be retained for at least 1 year"
```
Comment (gruebel) [inline checkov/terraform/checks/resource/aws/CloudWatchLogGroupRetentionYear.py:17]:
```suggestion
        name = "Ensure CloudWatch log groups retains logs for at least 1 year"
```

**Review by gruebel — COMMENTED** [review]
thanks for the contribution 🥇 , left a few minor comments

File: docs/5.Policy Index/all.md
Comment (quixoticmonk) [reply docs/5.Policy Index/all.md:1]: Done. Should have seen the auto-generated header :) .

**Review by ChanochShayner — APPROVED** [review]

**Review by gruebel — APPROVED** [review]

**Pull request merged**